designer/server/src/main/scala/pl/touk/nussknacker/ui/process/test/testdataformat/CommonDataFormatHandler.scala

```diff
@@ -50,6 +65,49 @@ class CommonDataFormatHandler(modelData: ModelData) extends TestDataFormatHandle
         Left(TestDataFormatHandler.LiveDataFetchingNotSupportedError)
     }
 
+  override def getTestParametersDefinition(
+      sourceId: NodeId,
+      sourceCompilationResult: NodeCompilationResult[Source]
+  ): Either[ScenarioTestService.ParametersDefinitionError, List[Parameter]] = {
+    for {
+      validOutputValidationContext <- sourceCompilationResult.validationContext.toEither.leftMap(errors =>
+        ParametersDefinitionError.SourcesCompilationError(NonEmptyList.one(sourceId -> errors))
+      )
+      recordTypeBasedOnOutputValidationContext = Typed.record(validOutputValidationContext.localVariables)
+      singleInputVariablesParameter = Parameter(InputVariablesParameterName, recordTypeBasedOnOutputValidationContext)
+        .copy(editors = List(JsonParameterEditor))
+    } yield List(singleInputVariablesParameter)
+  }
+
+  override def convertToTestData(
+      sourceId: String,
+      parameterExpressions: Map[ParameterName, Expression]
+  ): Either[TestDataFormatHandler.ExpressionsToTestDataConversionError, ScenarioTestData] = {
+    for {
+      inputVariablesExpression <- parameterExpressions
+        .get(InputVariablesParameterName)
+        .toRight(MissingInputVariablesParameter)
+      _ <- Either.cond(
+        inputVariablesExpression.language == Language.Json,
+        (),
+        IllegalInputVariablesExpressionLanguage(inputVariablesExpression.language)
+      )
+      validJson <- circe.parser
+        .parse(inputVariablesExpression.expression)
+        .leftMap(_.message)
+        .leftMap(InputVariablesExpressionJsonParseError)
+      validVariablesMap <- validJson
+        .as[Map[String, Json]]
+        .leftMap(_.message)
+        .leftMap(InputVariablesExpressionDecodingError)
+      singleRecord = ScenarioTestCommonFormatJsonRecord(NodeId(sourceId), validVariablesMap, timestamp = None)
```

Comment:
`timestamp = None` for now timestamp settings works only in case of tests from file?